rclcpp/test/rclcpp/test_rosout_qos.cpp

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <memory>
+
+#include "rclcpp/node_options.hpp"
+#include "rclcpp/qos.hpp"
+#include "rclcpp/duration.hpp"
+#include "rcl/logging_rosout.h"
+#include "rcl/init.h"
+#include "rmw/types.h"
+
+bool operator==(
+  const rmw_time_t & lhs,
+  const rmw_time_t & rhs)
+{
+  return lhs.sec == rhs.sec && lhs.nsec == rhs.nsec;
+}
+
+bool operator==(
+  const rmw_qos_profile_t & lhs,
+  const rmw_qos_profile_t & rhs)
+{
+  return lhs.history == rhs.history &&
+         lhs.depth == rhs.depth &&
+         lhs.reliability == rhs.reliability &&
+         lhs.durability == rhs.durability &&
+         lhs.deadline == rhs.deadline &&
+         lhs.lifespan == rhs.lifespan &&
+         lhs.liveliness == rhs.liveliness &&
+         lhs.liveliness_lease_duration == rhs.liveliness_lease_duration &&
+         lhs.avoid_ros_namespace_conventions == rhs.avoid_ros_namespace_conventions;
+}
+
+bool operator!=(
+  const rmw_qos_profile_t & lhs,
+  const rmw_qos_profile_t & rhs)
+{
+  return !(lhs == rhs);
+}
+
+class TestRosoutQoS : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    node_options = std::make_shared<rclcpp::NodeOptions>();
+  }
+
+  void InitializeNodeOptions(const rmw_qos_profile_t rmw_qos_options = rmw_qos_profile_default)
+  {
+    rclcpp::QoS qos_profiles(rmw_qos_options);
+    node_options->rosout_qos(qos_profiles);
+  }
+
+  void TearDown()
+  {
+    node_options.reset();
+  }
+
+  std::shared_ptr<rclcpp::NodeOptions> node_options;
+};
+
+/*
+   Test rosout_qos function with default value.
+ */
+TEST_F(TestRosoutQoS, test_rosout_qos_with_default_value) {
```

Comment:
If this is not using the fixture, it should be `TEST`:
```suggestion
TEST(TestRosoutQoS, test_rosout_qos_with_default_value) {
```

Reply:
Ok